earth2studio/models/dx/cbottle_tc.py

```diff
@@ -0,0 +1,539 @@
+        seed : int, optional
+            Random generator seed for latent variables, by default 0
+
+        Returns
+        -------
+        DiagnosticModel
+            Diagnostic model
+        """
+        checkpoints = [
+            package.resolve("cBottle-3d/training-state-000512000.checkpoint"),
+            package.resolve("cBottle-3d/training-state-002048000.checkpoint"),
+            package.resolve("cBottle-3d/training-state-009856000.checkpoint"),
+        ]
+        # https://github.com/NVlabs/cBottle/blob/4f44c125398896fad1f4c9df3d80dc845758befa/src/cbottle/inference.py#L106
+        core_model = MixtureOfExpertsDenoiser.from_pretrained(
+            checkpoints, (100.0, 10.0)
+        )
+
+        classifier_model = None
+        with Checkpoint(
+            package.resolve("cBottle-3d-tc/training-state-002176000.checkpoint")
+        ) as c:
+            classifier_model = c.read_model().eval()
+
+        sst_ds = xr.open_dataset(
+            package.resolve("amip_midmonth_sst.nc"),
+            engine="h5netcdf",
+            storage_options=None,
+            cache=False,
+        ).load()
+
+        return cls(
+            core_model,
+            classifier_model,
+            sst_ds,
+            sampler_steps=sampler_steps,
+            sigma_max=sigma_max,
+            seed=seed,
+        )
+
+    @staticmethod
+    def create_guidance_tensor(
+        lat_coords: torch.Tensor,
+        lon_coords: torch.Tensor,
+        times: list[datetime] | TimeArray,
+    ) -> tuple[torch.Tensor, OrderedDict]:
+        """Creates a TC guidance tensor from lat/lon coordinates.
+
+        Parameters
+        ----------
+        lat_coords : torch.Tensor
+            Latitude coordinates where TC guidance should be set
+        lon_coords : torch.Tensor
+            Longitude coordinates where TC guidance should be set
+        times: list[datetime] | TimeArray
+            List of datetime objects or numpy datetime64 array specifying the times for
+            the guidance tensor's coordinate system
+
+        Returns
+        -------
+        tuple[torch.Tensor, OrderedDict]
+            Tuple containing:
+            - Guidance tensor with shape (1,n,1,721,1440) where values are 1 at the
+                specified coordinates
+            - OrderedDict with coordinate dimensions and values
+        """
+        times = to_time_array(times)
+        # Convert any longitudes in -180 to 180 range to 0 to 360 range
+        lon_coords = torch.where(lon_coords < 0, lon_coords + 360, lon_coords)
+
+        lat_grid = np.linspace(90, -90, 721)
+        lon_grid = np.linspace(0, 360, 1440, endpoint=False)
+        guidance = torch.full((1, times.shape[0], 1, 721, 1440), 0).float()
+
+        lat_idx = torch.searchsorted(torch.from_numpy(-lat_grid), -lat_coords)
+        lon_idx = torch.searchsorted(torch.from_numpy(lon_grid), lon_coords)
+        guidance[0, 0, 0, lat_idx, lon_idx] = 1
```

Comment:
Honestly pretty confused how all this works but should guidance be effected by times as well? Like set it for all times not just the first one.